core/src/machine/step.rs

```diff
@@ -414,6 +530,40 @@ impl Machine {
             opcode!("CP A")    => cp!(self.cpu.a),
             opcode!("CP d8")   => cp!(arg_byte),
 
+            // ========== RST ==========
+            opcode!("RST 00H") => {
+                self.push(self.cpu.pc);
+                self.cpu.pc = Word::new(0x00);
+            }
+            opcode!("RST 08H") => {
+                self.push(self.cpu.pc);
+                self.cpu.pc = Word::new(0x08);
+            }
+            opcode!("RST 10H") => {
+                self.push(self.cpu.pc);
+                self.cpu.pc = Word::new(0x10);
+            }
+            opcode!("RST 18H") => {
+                self.push(self.cpu.pc);
+                self.cpu.pc = Word::new(0x18);
+            }
+            opcode!("RST 20H") => {
+                self.push(self.cpu.pc);
+                self.cpu.pc = Word::new(0x20);
+            }
+            opcode!("RST 28H") => {
+                self.push(self.cpu.pc);
+                self.cpu.pc = Word::new(0x28);
+            }
+            opcode!("RST 30H") => {
+                self.push(self.cpu.pc);
+                self.cpu.pc = Word::new(0x30);
+            }
+            opcode!("RST 38H") => {
+                self.push(self.cpu.pc);
+                self.cpu.pc = Word::new(0x38);
+            }
```

Comment:
I guess a `call!` macro would be handy here. This would work for all RST and all CALL instructions.